modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-builder/src/main/resources/META-INF/resources/js/components/LayoutProvider/LayoutProvider.es.js

```diff
@@ -432,6 +433,16 @@ class LayoutProvider extends Component {
 		});
 	}
 
+	_handlePagesSwapped({firstIndex, secondIndex}) {
+		const {pages} = this.state;
+
+		[pages[firstIndex], pages[secondIndex]]= [pages[secondIndex], pages[firstIndex]];
```

Comment:
Super genius!

Although, as a good practice, we usually try to avoid changing by reference and instead do it in the immutable way (See [this](https://redux.js.org/recipes/structuring-reducers/immutable-update-patterns#inserting-and-removing-items-in-arrays)). 
I don't see a problem here, just trying to enforce some standard as to what the other action handlers in this class are doing. This could be refactored to something like:

```js
const [firstPage, secondPage] = [pages[firstIndex], pages[secondIndex]];

this.setState({
    pages: pages.map((page, index) => index === firstIndex ?
        secondPage : index === secondIndex ? firstPage : page)
```
});

or just use `if/else` statements for clarity:

```js
const [firstPage, secondPage] = [pages[firstIndex], pages[secondIndex]];

this.setState({
    pages: pages.map((page, index) => {
        if (index === firstIndex) {
            return secondPage
        } else if (index === secondIndex) {
            return firstPage;
        }
        return page;
    })
}
```

Reply:
Hi, Timão. Thanks for the orientations! I'll make the change.